pkgs/by-name/om/omada-sdn-controller/package.nix

```diff
@@ -0,0 +1,149 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+  makeWrapper,
+  writeScriptBin,
+  coreutils,
+  curl,
+  jsvc,
+  openjdk17,
+  mongodb-ce,
+  omada-sdn-controller,
+  testers,
+  writeShellScript,
+  common-updater-scripts,
+  pup,
+  gnugrep,
+  gnused,
+  dataDir ? "/var/lib/omada",
+}:
+
+let
+  # The 'arch' command is required during runtime with is equivalent to 'uname -m':
+  arch = writeScriptBin "arch" "uname -m";
+
+  # Java 17 is supported, pin to avoid unsupported dependency upgrades.
+  jre = openjdk17;
+
+  # MongoDB 7 is supported with version 5.14.20 and above, pin to avoid unsupported dependency upgrades.
+  mongodb = mongodb-ce.overrideAttrs (oldAttrs: {
+    version = "7.0.14";
+  });
+in
+
+stdenv.mkDerivation rec {
+  pname = "omada-sdn-controller";
+  version = "5.14.26.1";
+
+  src = fetchurl {
+    url = "https://static.tp-link.com/upload/software/2024/202407/20240710/Omada_SDN_Controller_v5.14.26.1_linux_x64.tar.gz";
+    hash = "sha256-hVQlbFTCcFoIbBDajCOdbg3m3OgXnIej3OBlbHL374E=";
+  };
+
+  nativeBuildInputs = [
+    makeWrapper
+  ];
+
+  runtimeDeps = [
+    arch
+    coreutils
+    curl
+    jre
+    jsvc
+  ];
+
+  unpackPhase = "tar -xzf $src";
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out
+    cp -r Omada_SDN_Controller_v${version}_linux_x64/* $out
+    patchShebangs $out
+
+    # Move away the property and data directories, storing them as defaults,
+    # and create links for them to the data directory
+    mv $out/properties $out/properties.defaults
+    ln -s ${dataDir}/properties $out/properties
+    mv $out/data $out/data.defaults
+    ln -s ${dataDir}/data $out/data
+
+    # Adjust the data directory to point to a writeable path
+    sed -i "s|^OMADA_HOME=.*|OMADA_HOME=\"${dataDir}\"|" $out/bin/control.sh
+
+    # Modify the file paths in properties to make sure they point to data directory
+    sed -i "s|\.\./|${dataDir}/|" $out/properties.defaults/omada.properties
+    sed -i "s|\.\./|${dataDir}/|" $out/properties.defaults/log4j2.properties
```

Comment:
Please use substituteInPlace with --replace-fail to notice when it no longer matches.